### ISSUE TITLE
Ubuntu ceserver crash.

### ISSUE THREAD

**Issue by LiamKarlMitchell** [cheat-engine/cheat-engine#32]
It crashes with a buffer overflow detection just after I accept to connect to it from windows cheat engine. Using ceserver downloaded from website.

accept=5
**\* buffer overflow detected ***: ./ceserver terminated
======= Backtrace: =========
/lib/x86_64-linux-gnu/libc.so.6(+0x7338f)[0x7f58acaa338f]
/lib/x86_64-linux-gnu/libc.so.6(__fortify_fail+0x5c)[0x7f58acb3ac9c]
/lib/x86_64-linux-gnu/libc.so.6(+0x109b60)[0x7f58acb39b60]
./ceserver[0x40668f]
./ceserver[0x4078de]
./ceserver[0x408bf7]
/lib/x86_64-linux-gnu/libpthread.so.0(+0x8182)[0x7f58ad21a182]
/lib/x86_64-linux-gnu/libc.so.6(clone+0x6d)[0x7f58acb2a47d]
======= Memory map: ========
00400000-00410000 r-xp 00000000 fc:00 4762127                            /home/lmitchell/Downloads/ceserver
0060f000-00610000 r--p 0000f000 fc:00 4762127                            /home/lmitchell/Downloads/ceserver
00610000-00611000 rw-p 00010000 fc:00 4762127                            /home/lmitchell/Downloads/ceserver
00953000-00974000 rw-p 00000000 00:00 0                                  [heap]
7f58a4000000-7f58a4021000 rw-p 00000000 00:00 0 
7f58a4021000-7f58a8000000 ---p 00000000 00:00 0 
7f58ab818000-7f58ab82e000 r-xp 00000000 fc:00 11010053                   /lib/x86_64-linux-gnu/libgcc_s.so.1
7f58ab82e000-7f58aba2d000 ---p 00016000 fc:00 11010053                   /lib/x86_64-linux-gnu/libgcc_s.so.1
7f58aba2d000-7f58aba2e000 rw-p 00015000 fc:00 11010053                   /lib/x86_64-linux-gnu/libgcc_s.so.1
7f58aba2e000-7f58aba2f000 ---p 00000000 00:00 0 
7f58aba2f000-7f58ac22f000 rw-p 00000000 00:00 0                          [stack:19442]
7f58ac22f000-7f58ac230000 ---p 00000000 00:00 0 
7f58ac230000-7f58aca30000 rw-p 00000000 00:00 0                          [stack:19438]
7f58aca30000-7f58acbeb000 r-xp 00000000 fc:00 11015605                   /lib/x86_64-linux-gnu/libc-2.19.so
7f58acbeb000-7f58acdea000 ---p 001bb000 fc:00 11015605                   /lib/x86_64-linux-gnu/libc-2.19.so
7f58acdea000-7f58acdee000 r--p 001ba000 fc:00 11015605                   /lib/x86_64-linux-gnu/libc-2.19.so
7f58acdee000-7f58acdf0000 rw-p 001be000 fc:00 11015605                   /lib/x86_64-linux-gnu/libc-2.19.so
7f58acdf0000-7f58acdf5000 rw-p 00000000 00:00 0 
7f58acdf5000-7f58acdf8000 r-xp 00000000 fc:00 11014053                   /lib/x86_64-linux-gnu/libdl-2.19.so
7f58acdf8000-7f58acff7000 ---p 00003000 fc:00 11014053                   /lib/x86_64-linux-gnu/libdl-2.19.so
7f58acff7000-7f58acff8000 r--p 00002000 fc:00 11014053                   /lib/x86_64-linux-gnu/libdl-2.19.so
7f58acff8000-7f58acff9000 rw-p 00003000 fc:00 11014053                   /lib/x86_64-linux-gnu/libdl-2.19.so
7f58acff9000-7f58ad011000 r-xp 00000000 fc:00 11014205                   /lib/x86_64-linux-gnu/libz.so.1.2.8
7f58ad011000-7f58ad210000 ---p 00018000 fc:00 11014205                   /lib/x86_64-linux-gnu/libz.so.1.2.8
7f58ad210000-7f58ad211000 r--p 00017000 fc:00 11014205                   /lib/x86_64-linux-gnu/libz.so.1.2.8
7f58ad211000-7f58ad212000 rw-p 00018000 fc:00 11014205                   /lib/x86_64-linux-gnu/libz.so.1.2.8
7f58ad212000-7f58ad22b000 r-xp 00000000 fc:00 11015607                   /lib/x86_64-linux-gnu/libpthread-2.19.so
7f58ad22b000-7f58ad42a000 ---p 00019000 fc:00 11015607                   /lib/x86_64-linux-gnu/libpthread-2.19.so
7f58ad42a000-7f58ad42b000 r--p 00018000 fc:00 11015607                   /lib/x86_64-linux-gnu/libpthread-2.19.so
7f58ad42b000-7f58ad42c000 rw-p 00019000 fc:00 11015607                   /lib/x86_64-linux-gnu/libpthread-2.19.so
7f58ad42c000-7f58ad430000 rw-p 00000000 00:00 0 
7f58ad430000-7f58ad453000 r-xp 00000000 fc:00 11014185                   /lib/x86_64-linux-gnu/ld-2.19.so
7f58ad622000-7f58ad626000 rw-p 00000000 00:00 0 
7f58ad64e000-7f58ad652000 rw-p 00000000 00:00 0 
7f58ad652000-7f58ad653000 r--p 00022000 fc:00 11014185                   /lib/x86_64-linux-gnu/ld-2.19.so
7f58ad653000-7f58ad654000 rw-p 00023000 fc:00 11014185                   /lib/x86_64-linux-gnu/ld-2.19.so
7f58ad654000-7f58ad655000 rw-p 00000000 00:00 0 
7ffff4bfb000-7ffff4c1c000 rw-p 00000000 00:00 0                          [stack]
7ffff4ddc000-7ffff4dde000 r-xp 00000000 00:00 0                          [vdso]
ffffffffff600000-ffffffffff601000 r-xp 00000000 00:00 0                  [vsyscall]

**Comment by LiamKarlMitchell** [comment]
I saw there is one in the repo so I tried that.

https://github.com/cheat-engine/cheat-engine/blob/master/Cheat%20Engine/ceserver/Release-linux/ceserver

It gets further.

A bunch of text in the console on linux, and in windows I get an Access Violation error messagebox.
The error this time is

```
Failed to open [vdso]
```

Is there a specific combination of client/server versions I must use.
Or maybe compile them my self?
